### ISSUE TITLE
camelcase ignoreDestructuring doesn't work for objects inside objects

### ISSUE THREAD

**Issue by brizental** [eslint/eslint#11228]
**Tell us about your environment**
- ESLint Version: 5.11.1
- Node Version: 8.11.4
- npm Version: 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
        "camelcase": [
            "error",
            {
                "ignoreDestructuring": true
            }
        ],
        "indent": [
            "error",
            4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "never"
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
```
const obj = {
    property: {
        nested_property: "what"
    }
}
const { property: { nested_property } } = obj
```
```
eslint .
```

**What did you expect to happen?**
`nested_property` to be ignored

**What actually happened? Please include the actual, raw output from ESLint.**
```
3:9   error  Identifier 'nested_property' is not in camel case     camelcase
```

**Are you willing to submit a pull request to fix this bug?**
Sure

**Comment by platinumazure** [comment]
Hi @brizental, thanks for the issue.

Looking at the error message, it seems that line 3 is being reported. In your example, line 3 is an object property assignment, not a destructure.

If you want the rule to ignore property assignments that are not in camelcase, you should use the [`properties: "never"`](https://eslint.org/docs/rules/camelcase#properties-never) option, in addition to the `ignoreDestructuring` option you are already using in your configuration (and which successfully prevented a lint message from appearing on line 6).

With that said, I believe the rule is working as intended with your current configuration and example. Please reply if you feel I have missed something. Thanks!

**Comment by brizental** [comment]
Alright, so this is my bad... I tried to do a super stripped down example of the bug I found and ended up losing it, lol.

So this is the problem I actually had, didn't realize it only happened for arguments:

```javascript
const obj = {
    property: {
        nested_property: "what"
    }
}

function test({ property: { nested_property } }) {
    return nested_property
}

console.log(test(obj))
```

The error I get and shouldn't (or at least I think I shouldn't):

```
7:12  error  Identifier 'nested_property' is not in camel case  camelcase
```

I even had `"properties": "never"` set in my real code... hahaha.

Anyways, this looks like a duplicate of #11226 or at least something that is related to it. If it is found to be a bug I would still be willing to send in a PR.

Let me know and sorry for the confusion!
